.github/workflows/functional-test.yaml

```diff
@@ -94,22 +98,27 @@ jobs:
             echo "CHECKOUT_REPO=${{ github.repository }}" >> $GITHUB_ENV
             echo "CHECKOUT_REF=${{ github.ref }}" >> $GITHUB_ENV
             echo "PR_NUMBER=${{ github.event.pull_request.number }}" >> $GITHUB_ENV
-      - name: Set up checkout target (repository_dispatch from /ok-to-test)
-        if: github.event_name == 'repository_dispatch'
+      - name: Use custom actions
+        uses: actions/checkout@v3
+      - name: 'Download PR data artifacts'
+        if: github.event_name == 'workflow_run'
+        uses: ./.github/actions/download-pr-data-artifact
```

Comment:
There's an alternative syntax for this reference: https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow

I don't think it will require the checkout, which might simplify things.

It's kinda hard to reason about the fact that we're checkout out the repo twice, it would be nice to avoid that complication.

Reply:
I am getting an error without this step. Also, I found this: 
https://github.com/orgs/community/discussions/26579

Reply:
I'm not suggesting you remove that step, and I don't think the discussion you linked is relevant.

You can use a local path as a reference, or you can use a github path.

For this example it would be:

```
uses: radius-project/radius/.github/workflows/download-pr-data-artifact@main
```

If you do it this way, you may not need the checkout. I think it functionally will do the same thing, but it might be less complicated. 

I'll leave this up to you whether it's better or not.

Reply:
I tried converting this composite action to reusable workflow but it is getting complicated with the download-artifact one. We cannot use the uses in a step...so need to convert the download part to a separate job. Added that as a dependency to the build job which is using the output set but that gives me an error about cyclic dependency: The workflow is not valid. .github/workflows/functional-test.yaml (Line: 80, Col: 12): Job 'build' depends on unknown job 'download_pr_number'. .github/workflows/functional-test.yaml (Line: 301, Col: 12): Job 'tests' depends on job 'build' which creates a cycle in the dependency graph. The trigger also needs to change since a reusable workflow requires a workflow_call trigger. I think it is easier to leave it as is with the composite actions.

Reply:
I'm not sure I understand what you're saying. 

My suggestion was to reference the reusable action using the repository-syntax instead of cloning the repo and using a relative file-path. Are you saying that's not possible?

Reply:
I am using a composite action right now: ./.github/actions/download-pr-data-artifact whereas https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow is about reusable workflows. Looks like composite actions require a repo checkout. Therefore, I tried converting the composite action into a reusable workflow.

Reply:
OK thanks, I think I was confused about the terminology.